packages/compiler-cli/ngcc/src/analysis/module_with_providers_analyzer.ts

```diff
@@ -122,47 +136,37 @@ export class ModuleWithProvidersAnalyzer {
     if (definition === null) {
       return null;
     }
+
+    // Get hold of the return statement expression for the function
     const body = definition.body;
     const lastStatement = body && body[body.length - 1];
-    const returnExpression =
-        lastStatement && ts.isReturnStatement(lastStatement) && lastStatement.expression || null;
-    const ngModuleProperty = returnExpression && ts.isObjectLiteralExpression(returnExpression) &&
-            returnExpression.properties.find(
-                prop =>
-                    !!prop.name && ts.isIdentifier(prop.name) && prop.name.text === 'ngModule') ||
-        null;
-
-    if (!ngModuleProperty || !ts.isPropertyAssignment(ngModuleProperty)) {
+    if (lastStatement === null || !ts.isReturnStatement(lastStatement) ||
+        lastStatement.expression == undefined) {
```

Comment:
but this one can't, so this could be

```suggestion
        lastStatement.expression === undefined) {
```